addons/survey/controllers/main.py

```diff
@@ -322,9 +322,9 @@ def _prepare_survey_data(self, survey_sudo, answer_sudo, **post):
                     answer_sudo.last_displayed_page_id.id if answer_sudo.last_displayed_page_id else 0)
 
                 if next_page_or_question:
-                    pages_or_questions = survey_sudo._get_pages_or_questions(answer_sudo)
-                    if pages_or_questions.ids.index(next_page_or_question.id) == len(pages_or_questions) - 1:
-                        data.update({'survey_last': True})
+                    data.update({
+                        'survey_last': survey_sudo._is_last_page_or_question(answer_sudo, next_page_or_question)
```

Comment:
Could be nice to replace other occurences where we want to know if it's the last question or not.
For example in `_prepare_manage_session_values`.

Maybe others I did not find?